util/tracing/detect/detect.go

```diff
@@ -106,39 +121,83 @@ func detect() error {
 		return err
 	}
 
-	sp := sdktrace.NewBatchSpanProcessor(exp)
+	// enable log with traceID when valid exporter
+	if texp != nil {
+		bklog.EnableLogWithTraceID(true)
 
-	if Recorder != nil {
-		Recorder.flush = sp.ForceFlush
+		sp := sdktrace.NewBatchSpanProcessor(texp)
+
+		if Recorder != nil {
+			Recorder.flush = sp.ForceFlush
+		}
+
+		sdktp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sp), sdktrace.WithResource(res))
+		closers = append(closers, sdktp.Shutdown)
+
+		exporter.SpanExporter = texp
+		tp = sdktp
+	}
+
+	var readers []sdkmetric.Reader
+	if mexp != nil {
+		// Create a new periodic reader using any configured metric exporter.
+		readers = append(readers, sdkmetric.NewPeriodicReader(mexp))
 	}
 
-	sdktp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sp), sdktrace.WithResource(res))
-	closers = append(closers, sdktp.Shutdown)
+	if r, err := prometheus.New(); err == nil {
+		// Register the prometheus reader if there was no error.
+		// Don't fail if there was an error.
+		// TODO(jsternberg): Maybe log the error here?
```

Comment:
Yes I think we should log the error using `github.com/moby/buildkit/util/bklog`

Reply:
Forgot that I put this todo there. Good catch. I'll log the error as a warning.